exercises/practice/ledger/.meta/example.cairo

```diff
@@ -0,0 +1,224 @@
+// Refactored code
+use core::to_byte_array::AppendFormattedToByteArray;
+
+#[derive(Debug, PartialEq, Drop)]
+pub enum Currency {
+    USD,
+    EUR,
+}
+
+#[derive(Debug, PartialEq, Drop)]
+pub enum Locale {
+    en_US,
+    nl_NL,
+}
+
+pub fn format_entries(
+    currency: Currency, locale: Locale, transactions: Array<(ByteArray, ByteArray, ByteArray)>
+) -> Array<ByteArray> {
+    let mut ledger: Array<ByteArray> = array![];
+
+    // Step 1: Define the header based on the locale
+    let header = match @locale {
+        Locale::en_US => "Date       | Description               | Change       ",
+        Locale::nl_NL => "Datum      | Omschrijving              | Verandering  ",
+    };
+    ledger.append(header);
+
+    // Step 2: Process transactions
+    for (
+        date, transaction, change
+    ) in transactions {
+        let formatted_date = format_date(@date, @locale);
+
+        // Format the change based on the currency and locale
+        let formatted_change = format_change(@change, @currency, @locale);
+
+        // Format the transaction row
+        let mut row = formatted_date;
+        row += " | ";
+        row += format_transaction(transaction);
+        row += " | ";
+        row += formatted_change;
+
+        // Append the row to the ledger
+        ledger.append(row);
+    };
+
+    ledger
+}
+
+// format date based on the locale handled in 1 function
+fn format_date(date: @ByteArray, locale: @Locale) -> ByteArray {
+    let (mut year, mut month, mut day) = split_date(date);
+    match locale {
+        Locale::en_US => {
+            day += "/";
+            month += "/";
+            ByteArrayTrait::concat(@month, @ByteArrayTrait::concat(@day, @year))
+        },
+        Locale::nl_NL => {
+            day += "-";
+            month += "-";
+            ByteArrayTrait::concat(@day, @ByteArrayTrait::concat(@month, @year))
+        },
+    }
+}
+
+// split date into year, month and day
+fn split_date(date: @ByteArray) -> (ByteArray, ByteArray, ByteArray) {
+    let mut year = "";
+    let mut month = "";
+    let mut day = "";
+    let mut sep = 0;
+    let mut i = 0;
+
+    while i < date.len() {
```

Comment:
You could probably separate this single `while` loop into 3 loops, each building it's own part of the date